### PR TITLE
Build: split `lib/addons` into manager & preview part, move code to `manager-api`

### PR DESCRIPTION
**Pull request by ndelangen**
Move preview related code from `lib/addons` partially into `preview-api`

before:
Untitled Diagram drawio (6)

after:
storybook-prebundling-preview-architecture drawio

I've also moved all manager related code from`lib/addons` to manager-api:
storybook-prebundling-preview-architecture drawio (1)

`lib/addons` still exist, but it just re-exports parts from `preview-api` and `manager-api`.
This is for backwards compatibility

### CODE REVIEW

**Comment by ndelangen** [comment]
@JReinhold I think I'd like to verify with you something that seems to be happening here:
https://www.chromatic.com/test?appId=635781f3500dd2c49e189caf&id=637feadb2547552d19290761

When I load this story locally, the play function is ran, and the component renders as the before did in chromatic:
Screenshot 2022-11-24 at 23 37 53

But it seems in chromatic the 'undefined' state is stuck, which would mean the play function did not run?
Screenshot 2022-11-24 at 23 38 12

but somehow the play-function DID run on other stories, or we'd see many more stories get flagged for review.
So I'm wondering if this is some timing issue that is surfacing, or what else could be the cause?
Any ideas @JReinhold or @shilman ? 🙏

**Comment by ndelangen** [comment]
I'll see if i can add a delay to the story, perhaps chromatic doesn't understand that this is actually a story with a play function at all, therefore it doesn't wait for the play function to finish.

**Comment by JReinhold** [comment]
The special case about that story is that in itself it doesn't contain any interaction tests, but it contains a `<Story />` block which references a story that **does** have interaction tests, which it is suppose to run because of `docs.autoplay` on that sub-story.

So Chromatic doesn't know there's an interaction test to run here. We should just add a delay that ensure the nested interaction test finishes.

File: code/addons/docs/docs/multiframework.md [hunk 1]
```diff
@@ -112,6 +112,8 @@ This dynamic rendering is framework-specific, meaning it needs a custom implemen
 
 Let's take a look at the React framework implementation of `dynamic` snippets as a reference for implementing this feature in other frameworks:
 
+TODO: understanding this better, should probably import from `preview-api` instead of `addons`?
+
 ```tsx
 import { addons, StoryContext } from '@storybook/addons';
```
Comment (ndelangen) [inline code/addons/docs/docs/multiframework.md:118]: I do not know where this code runs, is this manager or preview code?

Comment (ndelangen) [inline code/addons/docs/docs/multiframework.md:118]:
```suggestion
```tsx
import { addons, StoryContext } from '@storybook/preview-api';
```

File: code/lib/channels/README.md [hunk 2]
```diff
@@ -31,6 +31,8 @@ class Transport {
 
 Currently, channels are baked into storybook implementations and therefore this module is not designed to be used directly by addon developers. When developing addons, use the `getChannel` method exported by `@storybook/addons` module. For this to work, Storybook implementations should use the `setChannel` method before loading addons.
 
+TODO: this needs to be updated to reflect the manager/preview split
+
 ```js
 import { addons } from '@storybook/addons';
```
Comment (ndelangen) [inline code/lib/channels/README.md:38]: I'm going to delete this code, we should document how to get the channel in the preview and manager somewhere @jonniebigodes, do we have a good place for that?

File: code/lib/preview/src/globals/exports.ts [hunk 3]
```diff
@@ -8,10 +8,8 @@ export default {
     'HooksContext',
     'addons',
     'applyHooks',
-    'isSupportedType',
     'makeDecorator',
     'mockChannel',
-    'types',
```
Comment (ndelangen) [inline code/lib/preview/src/globals/exports.ts:14]: might be a regression, should investigate

File: code/ui/.storybook/manager.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { addons } from '@storybook/addons';
```
Comment (ndelangen) [inline code/ui/.storybook/manager.ts:1]: add an entry to migration.md

**Review by yannbf — APPROVED** [review]
LGTM! Make sure to remember about documentation bits!

**Pull request merged**